# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: /home/jb198685/NetBeansProjects/jakarta-jmeter-src/src/core
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: org/apache/jmeter/gui/GuiPackage.java
--- org/apache/jmeter/gui/GuiPackage.java Base (BASE)
+++ org/apache/jmeter/gui/GuiPackage.java Locally Modified (Based On LOCAL)
@@ -619,7 +619,18 @@
 		return testPlanFile;
 	}
 
+        public void clearTestPlan() {
+            getTreeModel().clearTestPlan();
+            nodesToGui.clear();
+            setTestPlanFile(null);
+        }
 
+        public void clearTestPlan(TestElement element) {
+            getTreeModel().clearTestPlan(element);
+            removeNode(element);
+        }
+
+
\ No newline at end of file
     public static void showErrorMessage(final String message, final String title){
         showMessage(message,title,JOptionPane.ERROR_MESSAGE);
     }
Index: org/apache/jmeter/gui/action/Close.java
--- org/apache/jmeter/gui/action/Close.java Base (BASE)
+++ org/apache/jmeter/gui/action/Close.java Locally Modified (Based On LOCAL)
@@ -96,7 +96,7 @@
 	static void closeProject(ActionEvent e) {
 		GuiPackage guiPackage = GuiPackage.getInstance();
 
-		guiPackage.getTreeModel().clearTestPlan();
+		guiPackage.clearTestPlan();
 		guiPackage.getTreeListener().getJTree().setSelectionRow(1);
 
 		// Clear the name of the test plan file
Index: org/apache/jmeter/gui/action/Load.java
--- org/apache/jmeter/gui/action/Load.java Base (BASE)
+++ org/apache/jmeter/gui/action/Load.java Locally Modified (Based On LOCAL)
@@ -146,7 +146,7 @@
 		// If we are loading a new test plan, initialize the tree with the testplan node we are loading
 		GuiPackage guiInstance = GuiPackage.getInstance();
 		if(isTestPlan && !merging) {
-			guiInstance.getTreeModel().clearTestPlan((TestElement)tree.getArray()[0]);
+			guiInstance.clearTestPlan((TestElement)tree.getArray()[0]);
 		}
 
 		if (merging){ // Check if target of merge is reasonable
Index: org/apache/jmeter/reporters/AbstractListenerElement.java
--- org/apache/jmeter/reporters/AbstractListenerElement.java Base (BASE)
+++ org/apache/jmeter/reporters/AbstractListenerElement.java Locally Modified (Based On LOCAL)
@@ -18,6 +18,7 @@
 
 package org.apache.jmeter.reporters;
 
+import java.lang.ref.WeakReference;
 import org.apache.jmeter.testelement.AbstractTestElement;
 import org.apache.jmeter.visualizers.Visualizer;
 
@@ -27,17 +28,17 @@
  */
 
 public abstract class AbstractListenerElement extends AbstractTestElement {
-	transient private Visualizer listener;
+	transient private WeakReference listener;
 
 	public AbstractListenerElement() {
 	}
 
 	protected Visualizer getVisualizer() {
-		return listener;
+		return (Visualizer)listener.get();
 	}
 
 	public void setListener(Visualizer vis) {
-		listener = vis;
+		listener = new WeakReference(vis);
 	}
 
 	public Object clone() {
